src/main/java/com/dev/movietheatre/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.dev.movietheatre.service.impl;
+
+import com.dev.movietheatre.dao.ShoppingCartDao;
+import com.dev.movietheatre.dao.TicketDao;
+import com.dev.movietheatre.lib.Inject;
+import com.dev.movietheatre.lib.Service;
+import com.dev.movietheatre.model.MovieSession;
+import com.dev.movietheatre.model.ShoppingCart;
+import com.dev.movietheatre.model.Ticket;
+import com.dev.movietheatre.model.User;
+import com.dev.movietheatre.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCartByUser = shoppingCartDao.getByUser(user);
```

Comment:
we have the opportunity to pass to this method the user who is not saved to the DB (or who does not have the shopping cart - admin for example)

if so, your method will save the ticket to the DB anyway, even if on the next line there will be no cart for the user

let's fix this

Reply:
i'm not sure my simple fix is what you required. Maybe your required more complicated fix.